source/GameInterface.Tests/Serialization/SerializerTests/CharacterPerksSerializationTest.cs

```diff
@@ -0,0 +1,83 @@
+﻿using GameInterface.Serialization;
+using GameInterface.Serialization.Impl;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Serialization;
+using TaleWorlds.CampaignSystem.CharacterDevelopment;
+using TaleWorlds.Core;
+using TaleWorlds.ObjectSystem;
+using Xunit;
+
+namespace GameInterface.Tests.Serialization.SerializerTests
+{
+    public class CharacterPerksSerializationTest
+    {
+        public CharacterPerksSerializationTest()
+        {
+            MBObjectManager.Init();
+            MBObjectManager.Instance.RegisterType<PerkObject>("PerkObject", "PerkObjects", 4U, true, false);
+        }
+        [Fact]
+        public void CharacterPerks_Serialize()
+        {
+            CharacterPerks CharacterPerks = new CharacterPerks();
+
+            BinaryPackageFactory factory = new BinaryPackageFactory();
+            CharacterPerksBinaryPackage package = new CharacterPerksBinaryPackage(CharacterPerks, factory);
+
+            package.Pack();
+
+            byte[] bytes = BinaryFormatterSerializer.Serialize(package);
+
+            Assert.NotEmpty(bytes);
+        }
+
+        private static readonly FieldInfo _attributes = typeof(PropertyOwner<PerkObject>).GetField("_attributes", BindingFlags.NonPublic | BindingFlags.Instance);
+        [Fact]
+        public void CharacterPerks_Full_Serialization()
+        {
+            CharacterPerks characterPerks = new CharacterPerks();
+
+            characterPerks.StringId = "myCharacterPerks";
+
+            MBObjectManager.Instance.RegisterObject(characterPerks);
```

Comment:
Register the perk objects on lines 48-51, anything that is resolved using MBObjectManager.Instance.GetObject